drivers/hmis/app/graphql/types/hmis_schema/ce_referral.rb

```diff
@@ -10,43 +10,70 @@ module Types
+    summary_field :created_at, GraphQL::Types::ISO8601DateTime, null: false
+    summary_field :source_enrollment, Types::HmisSchema::CeReferralSourceEnrollment, null: true, description: 'Limited details about the source enrollment. Available even without full access to the source record.'
+    # Resolve project fields separately, instead of on the project schema object, in case user can't view the project
+    summary_field :target_project_id, ID, null: false
+    summary_field :target_project_name, String, null: false
+    summary_field :target_project_type, HmisSchema::Enums::ProjectType, null: false
+    summary_field :target_organization_name, String, null: false
+
+    summary_field :referred_by, Application::User, null: true
+    summary_field :active, Boolean, null: false, method: :active?
+
+    access_field authorize_with: nil do
+      field :can_view_referral_details, Boolean, null: false
+      field :can_view_target_project, Boolean, null: false
+    end
 
+    # Detailed fields that only those with full view access should see. Must be nullable
+    field :opportunity, HmisSchema::CeOpportunity, null: true
+    field :steps, [HmisSchema::CeReferralStep], null: true
+    field :client_name, String, null: true, description: 'The name of the referred client. Always available to those who can view the referral, even without full client record access.'
+    field :client_age, Integer, null: true, description: 'The age of the referred client. Always available to those who can view the referral, even without full client record access.'
     field :current_steps, [HmisSchema::CeReferralStep], null: true
     field :days_on_current_steps, Integer, null: true
     field :updated_by, Application::User, null: true
     field :target_enrollment, Types::HmisSchema::Enrollment, null: true, description: 'Target enrollment, if the user has permission to view it.'
-    field :source_enrollment, Types::HmisSchema::CeReferralSourceEnrollment, null: true, description: 'Limited details about the source enrollment. Available even without full access to the source record.'
-    field :swimlanes, [HmisSchema::CeReferralSwimlane], null: false
+    field :swimlanes, [HmisSchema::CeReferralSwimlane], null: true
     field :workflow_template_name, String, null: true
-    field :audit_events, HmisSchema::CeReferralAuditEvent.page_type, null: false
-    field :notes, HmisSchema::CeReferralNote.page_type, null: false
-
+    field :audit_events, HmisSchema::CeReferralAuditEvent.page_type, null: true
+    field :notes, HmisSchema::CeReferralNote.page_type, null: true
     # generically resolve current values for any fields referenced by Match Rule expressions
     field :current_match_values, [HmisSchema::CeMatchValue], null: true, description: 'Eligibility-related field values. May expose data beyond normal permissions.', method: :resolve_match_rule_fields
 
```

Comment:
I think we need to add referral origin and the ability to filter on that? Sorry if that is captured by a different PR or issue I may have missed it. I'm thinking of the CE Admin page in particular, which should support filtering out direct-sent referrals.

Reply:
Sounds good sorry for missing it! https://github.com/greenriver/hmis-warehouse/pull/5589/commits/90ed3dca49b2ebfb9c95336234ffbf6b40eb6721